File: day1/cs/Program.cs
var maxList = parseFile();

Console.WriteLine($"Day 1-1 : {maxList.Max()}");
Console.WriteLine($"Day 1-2 : {maxList.OrderDescending().Take(3).Sum()}");

IList<int> parseFile()
{
    var lines = File.ReadAllText(@"..\input1.txt");

    var groups = lines.Split(Environment.NewLine + Environment.NewLine);
    List<int> maxList = new();
    foreach(var group in groups)
    {
        maxList.Add(group.Split("\n").Select(x => int.Parse(x)).Sum());
    }
    return maxList;
}

File: day1/fsharp/Program.fs
open System

let input = System.IO.File.ReadAllText @"..\input1.txt"
// let input = @"1000
// 2000
// 3000

// 4000

// 5000
// 6000

// 7000
// 8000
// 9000

// 10000"

let calc (inp:string) =
    let sumGroup (grp:string) =
        grp.Split Environment.NewLine
        |> Seq.map int
        |> Seq.sum
    
    let groupTotals =
        inp.Split $"{Environment.NewLine}{Environment.NewLine}"
        |> Seq.map sumGroup

    printfn "Part 1 : %i" (groupTotals |> Seq.max)

    let part2 =
        groupTotals
        |> Seq.sortDescending
        |> Seq.take 3
        |> Seq.sum
    printfn "Part 2 : %i" part2

calc input